src/Storages/MergeTree/MergeTreeDataMergerMutator.cpp

```diff
@@ -372,6 +373,14 @@ bool MergeTreeDataMergerMutator::selectAllPartsToMergeWithinPartition(
         return false;
     }
 
+    /// If final, optimize_skip_merged_partitions is true and we have only one part in partition with level > 0
+    /// than we don't select it to merge
+    if (final && data.getSettings()->optimize_skip_merged_partitions && parts.size() == 1 && parts[0]->info.level > 0)
+    {
+        *is_single_merged_part = true;
+        return false;
+    }
+
```

Comment:
I think it's also necessary to check if there exists expired TTL. If so, merge of one part shall be run, because it will do some useful stuff. ([see](https://github.com/ClickHouse/ClickHouse/blob/7470ac98bb878e1b69ef5509cd9479e61dfafe5a/src/Storages/MergeTree/MergeTreeDataMergerMutator.cpp#L682-L697))

Reply:
I'll add this check, thanks